services/mailer/mailer.go

```diff
@@ -75,6 +83,18 @@ func (m *Message) SetHeader(field string, value ...string) {
 	m.Headers[field] = value
 }
 
+func (m *Message) generateMessageID() string {
+	var b bytes.Buffer
+	err := gob.NewEncoder(&b).Encode(m)
+	if err != nil {
+		log.Warn("Error converting mail message to bytes, falling back to randomness: %v", err)
+		uid, _ := uuid.NewRandom()
+		b.WriteString(uid.String())
+	}
+	now := time.Now().Format("20060102150405") // YYYYMMDDHHMMSS
```

Comment:
I think we do not need these complex methods. a `UUID` or `TimeMillSec + (4~8)RandomBytes` is enough.

Reply:
but isn't randomness slower than just use the message itself?

Reply:
How about `TimeMilSec + (len(m.To[0]) ^ len(m.Subject) ^ len(m.Body))`. That should be unique enough per message and does not need random numbers or hash calculation.

Reply:
seems good, changed to that